mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -627,3 +663,20 @@ def _construct_pipeline_component(  # pylint: disable=W0640
             raise TypeError(
                 f"{component} must be of type String or {expected_component_class.__name__}."
             )
+
+
+class TPPJSONEncoder(json.JSONEncoder):
+    """
+    Custom Encoder class defined to obtain recursive JSON representation of a TextPreparationPipeline.
+
+    Args:
+        None.
+
+    Returns:
+        Custom JSON Encoder class (json.JSONEncoder) .
+    """
+
+    def default(self, o):
```

Comment:
Do we want to handle the "else" case here where tojson is not callable?

Reply:
Fixed to throw a TextPreparationPipelineError for missing tojson method